### ISSUE TITLE
Casting required to setup replication to bucket in another stack

### ISSUE THREAD

**Issue by kierans** [rogerchi/cdk-s3-bucketreplication#211]
I have the setup where my source bucket is in one stack, and the destination bucket is in another stack.

To setup replication, I have to cast the "bucket from the other stack" to an `s3.Bucket` in order to have the code compile. For example

```typescript
const sourceBucket = new S3.Bucket(this, "SourceBucket", {});
const desttinationBucket = s3.Bucket.fromBucketName(this, "DestinationBucket", "other-bucket");

new BucketReplication(this, "BucketReplication", {
  sourceBucket,

  destinationBucket // does not compile

  destinationBucket: destinationBucket as s3.Bucket
})
```

This is because `fromBucketName` returns an instance of `s3.IBucket`. The replication still works with the cast. But it is ugly given the replication only needs the declared properties in `s3.IBucket`. It would be better to "code to the interface".

**Comment by kierans** [comment]
Thanks for merging this. Do you know when you'll be publishing the fix?

**Comment by rogerchi** [comment]
Hi, I just published 0.0.9 to npm.  Let me know if you have any issues, it's been a while since I looked at this repo.  I really appreciate the contribution!
